### DISCUSSION TITLE
My solution for filtering by the `belongsTo` relation

### DISCUSSION THREAD

**Discussion by rajabilal555** [orchidsoftware/platform#2210] [Show and tell]
Since there is no simple way of using the built in `HttpFilter` to filter relations, I have come up with a way to filter at least the basic `belongsTo` relations with out defining custom Filter classes...

Here is the basic code for it.
the `TD::make`'s name parameter should be same as the column in the db (in this case `sub_subject_id`).
If you have not too many Records in the "sub_subject" table like this example, then you can use the following:
```php
TD::make('sub_subject_id', 'Sub Subject')->render(function ($value) {
    return $value->sub_subject->name;
})->filter(TD::FILTER_SELECT, SubSubject::all()->pluck('name','id'))->sort(),
```
**Preview:** 
![image](https://user-images.githubusercontent.com/15360941/165009509-c1a0ad06-b53b-4a6b-9b20-61aaa770dd0b.png)

If you want Lazy Loaded dropdown and have a lot of records in the table then you can use the following:
```php
TD::make('sub_subject_id', 'Sub Subject')->render(function (ScreenModel $value) {
    return $value->sub_subject->name;
})->filter(Relation::make()->fromModel(SubSubject::class,'name'))->sort(),
```
The only problem with the Relation Field approach is that it will show the id instead of the name when filtering...
**Preview:**
![image](https://user-images.githubusercontent.com/15360941/165009446-63917049-6ac6-48ca-a030-17d17fea1cff.png)

I have not yet come up with a work around for the **Relation field Filter tag**, maybe some one can help with that... ?

**Caveats:** If you have a lot of Records you will probably have to go the Custom Filter Route as the current solution doesn't look as good.

**Comment by iFrez** [comment]
Hello, as a workaround you can override TD.php and modify  [buildFilterString() ]( https://github.com/orchidsoftware/platform/blob/d780d7feccaf85c1fc0a51ca48be524ff312f28c/src/Screen/TD.php#L384) to looks like     

>   if ($this->column == "column_name_id") {
            $Model = Model::whereId($filter)->first();
            return $Model === null ? $filter : $Model->name;
        }

Or if you use custom filter just adjust [value()](https://github.com/orchidsoftware/platform/blob/d780d7feccaf85c1fc0a51ca48be524ff312f28c/src/Filters/Filter.php#L122) function as you want.

**Comment by macit-emre** [comment]
use TD's filterValue() method for loading the right label for your selected filter. no need to override anything.
